config/v1/types_ingress.go

```diff
@@ -131,6 +131,7 @@ type LoadBalancer struct {
 
 // AWSIngressSpec holds the desired state of the Ingress for Amazon Web Services infrastructure provider.
 // This only includes fields that can be modified in the cluster.
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=SetEIPForNLBIngressController,rule="self.type == 'NLB' ? true : !has(self.networkLoadBalancerParameters)",message="Network Load Balancer parameters are allowed only when load balancer type is NLB."
```

Comment:
For consistency with other APIs, can you copy the style of please https://github.com/openshift/api/blob/4f6053f954b069f7fea5788d4307f4213db53678/example/v1/types_stable.go#L124C140-L124C199

Reply:
OK done